platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/CrashReporter.kt

```diff
@@ -0,0 +1,170 @@
+// capture-sdk - bitdrift's client SDK
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+package io.bitdrift.capture.reports
+
+import android.os.Build
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import io.bitdrift.capture.ContextHolder.Companion.APP_CONTEXT
+import io.bitdrift.capture.common.MainThreadHandler
+import io.bitdrift.capture.providers.FieldValue
+import io.bitdrift.capture.providers.toFieldValue
+import io.bitdrift.capture.reports.CrashReporterState.Initialized.ProcessingFailure
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.attribute.FileTime
+import kotlin.time.DurationUnit
+import kotlin.time.measureTime
+
+/**
+ * Handles internal reporting of crashes
+ */
+internal class CrashReporter(
+    private val mainThreadHandler: MainThreadHandler = MainThreadHandler(),
+) {
+    private val appContext = APP_CONTEXT
+
+    /**
+     * Process existing crash report files.
+     *
+     * @return The status of the crash reporting processing
+     */
+    fun processCrashReportFile(): CrashReporterStatus =
+        runCatching {
+            if (MainThreadHandler.isOnMainThread()) {
```

Comment:
nit: shouldn't this method be inside the instance of `mainThreadHandler` that way it could be easily mocked